### PR TITLE
Support encryption, decryption, and multi-dimensional accounts

### PR DESCRIPTION
**Pull request by carsonfarmer**
I'll make some comments to explain what I did here.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -3,20 +3,20 @@ import { BIP32 } from './bip32'
 import { Keypair } from './keypair'
 import { hmacSHA512 } from './crypto'
 
+export * from './keypair'
```
Comment (carsonfarmer) [inline src/index.ts]: This is a 'nice to have' for projects that depend on `js-wallet`

File: src/index.ts [hunk 2]
```diff
@@ -3,20 +3,20 @@ import { BIP32 } from './bip32'
...
+
 /**
  * A derived Wallet account
  * @property index The index for the given account
  * @property seed The Ed25519 private seed/key
  * @property address The Ed25519 public key
  */
 export interface Account {
-  index: number
+  index: number | number[]
```
Comment (carsonfarmer) [inline src/index.ts]: We now support multi-dimensional indices, but by default, will still support a single scalar index.

File: src/index.ts [hunk 3]
```diff
@@ -94,20 +94,29 @@ export default class Wallet {
...
+   * @returns A derived Wallet Account
    */
-  accountAt(index: number, password?: string): Account {
+  accountAt(index: number | number[], password?: string): Account {
     const seed = bip39.mnemonicToSeed(this.recoveryPhrase, password)
     const masterKey = createMasterKey(seed)
     const baseKey = masterKey.derivePath(TEXTILE_BIP44)
-    const accountKey = baseKey.deriveHardened(index)
+    const start = Array.isArray(index) ? index[0] : index
+    let accountKey = baseKey.deriveHardened(start)
```
Comment (carsonfarmer) [inline src/index.ts]: We'll always start with the first or indexth account. This is _probably_/by default the only one we want...

File: src/index.ts [hunk 4]
```diff
@@ -94,20 +94,29 @@ export default class Wallet {
...
+    if (Array.isArray(index) && index.length > 1) {
```
Comment (carsonfarmer) [inline src/index.ts]: But if we pass a multi-dimensional index, then we'll just loop through the remaining values in the array, treating each as an index 'deeper' into the account hierarchy.

File: src/keypair.ts [hunk 5]
```diff
@@ -123,4 +127,64 @@ export class Keypair {
...
+  /**
+   * Encrypts the given `data` using a Curve25519 'variant' of the public key.
+   *
+   * The encryption uses an ephemeral private key, which is prepended to the ciphertext,
+   * along with a nonce of random bytes.
+   *
+   * @note See https://github.com/dchest/ed2curve-js for conversion details.
+   * @param data Data to encrypt
+   */
+  encrypt(data: Buffer) {
```
Comment (carsonfarmer) [inline src/keypair.ts]: This is essentially a port of our `go-textile/crypto/asymmetric.go` `Encrypt` and `Decrypt` functions. We use `ed2curve` here, which is apparently compatible with `github.com/agl/ed25519/extra25519` (which is what we use).

File: src/index.ts (hunk 2 above)
Comment (sanderpick) [reply src/index.ts]: very cool

File: src/keypair.ts (hunk 5 above)
Comment (sanderpick) [reply src/keypair.ts]: wow, nice

**Pull request merged**